pymdownx/snippets.py

```diff
@@ -190,43 +195,65 @@ def download(self, url):
         The most recently used files are kept in a cache until the next reset.
         """
 
-        http_request = urllib.request.Request(url, headers=self.url_request_headers)
-        timeout = None if self.url_timeout == 0 else self.url_timeout
-        with urllib.request.urlopen(http_request, timeout=timeout) as response:
-
-            # Fail if status is not OK
-            status = response.status if util.PY39 else response.code
-            if status != 200:
-                raise SnippetMissingError(f"Cannot download snippet '{url}'")
-
-            # We provide some basic protection against absurdly large files.
-            # 32MB is chosen as an arbitrary upper limit. This can be raised if desired.
-            content = None
-            if "content-length" not in response.headers:
-                # we have to read to know if we went over the max, but never more than `url_max_size`
-                # where `url_max_size` == 0 means unlimited
-                content = response.read(self.url_max_size) if self.url_max_size != 0 else response.read()
-                content_length = len(content)
-            else:
-                content_length = int(response.headers["content-length"])
-
-            if self.url_max_size != 0 and content_length >= self.url_max_size:
-                raise ValueError(f"refusing to read payloads larger than or equal to {self.url_max_size}")
-
-            # Nothing to return
-            if content_length == 0:
-                return ['']
-
-            if content is None:
-                # content-length was in the header, so we did not read yet
-                content = response.read()
-
-            # Process lines
-            last = content.endswith((b'\r', b'\n'))
-            s_lines = [l.decode(self.encoding) for l in content.splitlines()]
-            if last:
-                s_lines.append('')
-            return s_lines
+        for attempt in range(1, self.max_retries + 1):
+            try:
+                http_request = urllib.request.Request(url, headers=self.url_request_headers)
+                timeout = None if self.url_timeout == 0 else self.url_timeout
+                with urllib.request.urlopen(http_request, timeout=timeout) as response:
+                    # Fail if status is not OK
+                    status = response.status if util.PY39 else response.code
+                    
+                    if status == 429:  # Rate limited
```

Comment:
This should be simplified as below which should handle all bad status.

```py
if status != 200:
    if status == 429 and attempt < self.max_retries:
        wait = self.backoff_factor * attempt
        time.sleep(wait)
        continue
    raise SnippetMissingError(f"Cannot download snippet '{url}'")
```

Additionally, if we are capturing the rate limited status here, why do we need to check for `urllib.error.HTTPError` later?